gpjax/__init__.py

```diff
@@ -13,19 +13,21 @@
 # limitations under the License.
 # ==============================================================================
 
-from .dataset import Dataset
-from .fit import fit
-from .gps import Prior, construct_posterior
-from .kernels import *
-from .likelihoods import Bernoulli, Gaussian
-from .mean_functions import Constant, Zero
-from .objectives import (ELBO, CollapsedELBO, ConjugateMLL,
-                         LogPosteriorDensity, NonConjugateMLL)
-from .variational_families import (CollapsedVariationalGaussian,
-                                   ExpectationVariationalGaussian,
-                                   NaturalVariationalGaussian,
-                                   VariationalGaussian,
-                                   WhitenedVariationalGaussian)
+from jaxtyping import install_import_hook
+with install_import_hook("gpjax", "beartype.beartype"):
```

Comment:
do we _want_ to push beartype onto end users?

Reply:
hm, mainly in some places we might have specified stricter types (array dtypes/shapes) than is strictly required by the code, so some code might have run fine if only beartype hadn't intervened. also, the explicit _check_shape error messages might be a bit more informative than the generic beartype ones. I don't know if there's also some more interaction with the jaxtyping `@jaxtyped` decorator.

Reply:
So if we didn't push beartype onto the end-user, then it would just be a testing utility for the package? I could easily imagine how any overly rigid beartype assertions could be annoying for an end user.

Reply:
Yeah! Though on the other hand, it could help us discover more bugs not handled by the tests...

Reply:
For example, a bunch of them were only uncovered by the notebooks. But I guess we could add beartype there as well, and then use that as a way to suggest to users that they _should_ use e.g. beartype (and report when one of our type hints is wrong)?

Reply:
That's a good way to position it. Let's push it onto users then. If it becomes an issue, we can always walk it back with little major disruption.

Reply:
Umm, do you mean _force_ it on users (by including it in the general import), or _strongly suggest it_ to users by having it at the start of every notebook (and inside the tests)? I think I'd be in favour of the latter actually...

Reply:
The latter - it's consistent with how we encourage people to use float64 without enforcing it.

Reply:
done